sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -970,6 +1052,26 @@ __ESIMD_API simd<Tx, N> atomic_update(Tx *p,
   return atomic_update<Op, Tx, N>(p, simd<Ty, N>(offsets), mask);
 }
 
+/// A variation of \c atomic_update API with \c offset represented as
+/// scalar.
+///
+/// @tparam Op The atomic operation - can be \c atomic_op::inc or
+/// \c atomic_op::dec, \c atomic_op::load.
+/// @tparam Tx The vector element type.
+/// @tparam N The number of memory locations to update.
+/// @param p The USM pointer.
+/// @param offset The scalar 32-bit or 64-bit offset in bytes.
```

Comment:
The comment here says it is 32 or 64-bit offset, while the condition on the line 1070 is 'std::is_integral_v<Toffset>. What happens if Toffset is uint16_t. Would the compilation pass?